src/Tools/dotnet-trace/CommandLine/Commands/ProvidersCommandHandler.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tracing.Eventing;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Diagnostics.Tracing;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Trace
+{
+    internal static class ProvidersCommandHandler
+    {
+        public static async Task<int> KnownProviders(IConsole console)
+        {
+            try
+            {
+                foreach (var provider in CommonProviders)
+                {
+                    var filterData = provider.FilterData == null ? "" : $":{provider.FilterData}";
+                    Console.Out.WriteLine($"Provider={provider.Name}:0x{provider.Keywords:X16}:{(uint)provider.EventLevel}{filterData}");
+                }
+
+                await Task.FromResult(0);
+                return 0;
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"[ERROR]: {ex.ToString()}");
+                return 1;
+            }
+        }
+
+        public static Command KnownProvidersCommand() =>
+            new Command(
+                name: "knownproviders",
+                description: "List known tracing flags.",
```

Comment:
I don't understand how you were intending this to be used? If I follow correctly this command is going to print some command-line output such as:

    Microsoft-Windows-DotNETRuntime:0x0000000000000001:3
    Microsoft-Windows-DotNETRuntime:0x0000000000000002:3
    Microsoft-Windows-DotNETRuntime:0x00000000FFFFFFFF:5
    ...

Reply:
This temporary for me for debugging. Option is hidden. Those are some of the providers that used to be default on xperf and I used for testing.